storage/range_test.go

```diff
@@ -1331,47 +1332,41 @@ func TestRangeIdempotence(t *testing.T) {
 	tc.Start(t)
 	defer tc.Stop()
 
-	// Run the same increment 100 times, 50 with identical command ID,
+	// Run the same increment 100 times, 50 with identical command IDs,
 	// interleaved with 50 using a sequence of different command IDs.
 	goldenArgs, _ := incrementArgs([]byte("a"), 1, 1, tc.store.StoreID())
-	incDones := make([]chan struct{}, 100)
+	const numIncs = 100
+	var wg sync.WaitGroup
 	var count int64
-	for i := range incDones {
-		incDones[i] = make(chan struct{})
-		idx := i
-		go func() {
-			var args proto.IncrementRequest
-			var reply proto.IncrementResponse
-			args = *goldenArgs
-			args.Header().Timestamp = tc.clock.Now()
-			if idx%2 == 0 {
-				args.CmdID = proto.ClientCmdID{WallTime: 1, Random: 1}
-			} else {
-				args.CmdID = proto.ClientCmdID{WallTime: 1, Random: int64(idx + 100)}
-			}
-
-			err := tc.rng.AddCmd(tc.rng.context(), proto.Call{Args: &args, Reply: &reply}, true)
+	incFunc := func(idx int) {
+		var args proto.IncrementRequest
+		var reply proto.IncrementResponse
+		args = *goldenArgs
+		args.Header().Timestamp = tc.clock.Now()
+		if idx%2 == 0 {
+			args.CmdID = proto.ClientCmdID{WallTime: 1, Random: 1}
+		} else {
+			args.CmdID = proto.ClientCmdID{WallTime: 1, Random: int64(idx + 100)}
+		}
+		if err := tc.rng.AddCmd(tc.rng.context(), proto.Call{Args: &args, Reply: &reply}, true); err != nil {
+			t.Fatal(err)
+		}
+		if idx%2 == 0 && reply.NewValue != 1 {
+			t.Errorf("expected all incremented values to be 1; got %d", reply.NewValue)
+		} else if idx%2 == 1 {
+			atomic.AddInt64(&count, reply.NewValue)
+		}
+		wg.Done()
+	}
 
-			if err != nil {
-				t.Fatal(err)
-			}
-			if idx%2 == 0 && reply.NewValue != 1 {
-				t.Errorf("expected all incremented values to be 1; got %d", reply.NewValue)
-			} else if idx%2 == 1 {
-				atomic.AddInt64(&count, reply.NewValue)
-			}
-			close(incDones[idx])
-		}()
+	wg.Add(numIncs)
+	incFunc(0) // run first iteration without concurrency.
```

Comment:
the assertions for the even-numbered half of this test look for a hardcoded value of `1`, so we have to make sure the first iteration that actually runs is even-numbered so that the cached response for that command ID has a value of 1. Otherwise, N odd-numbered operations may get scheduled first, run with a different command ID, and cause even-numbered operations to report something other than 1, depending on the value of N.